### PR TITLE
[BEAM-3999] Futurize internal subpackage

### PR DESCRIPTION
**Pull request by RobbeSneyders**
This pull request prepares the internal subpackage for Python 3 support. This PR is part of a series in which all subpackages will be updated using the same approach.
This approach has been documented [here](https://docs.google.com/document/d/1xDG0MWVlDKDPu_IW9gtMvxi2S9I0GB0VDTkPhjXT0nE) and the first pull request in the series (Futurize coders subpackage) demonstrating this approach can be found at #5053.

R: @aaltay @tvalentyn

### CODE REVIEW

**Comment by RobbeSneyders** [comment]
retest this please

File: sdks/python/apache_beam/internal/util.py [hunk 1]
```diff
@@ -20,9 +20,13 @@
 For internal use only. No backwards compatibility guarantees.
 """
 
+from __future__ import absolute_import
+
 import logging
 import threading
 import weakref
+from builtins import next
```
Comment (tvalentyn) [inline sdks/python/apache_beam/internal/util.py]: I'd like to understand the reason for adding `from builtins import next`. Is it  done by a conversion tool because we have an occurrence of `next(v_iter)` in the file, and the tool cannot infer whether or not v_iter implements a custom iterator? If yes: will such changes be required  by linter? 

Is my understanding correct that as long as v_iter's class does not implement a custom iterator (and it is not, in our case), the code without the import is still equivalent in Python 2 and Python 3?

**Comment by tvalentyn** [comment]
retest this please

**Comment by tvalentyn** [comment]
Let's see if the Py3 linter kicks in.

File: sdks/python/apache_beam/internal/util.py (hunk 1 above)
Comment (RobbeSneyders) [reply sdks/python/apache_beam/internal/util.py]: This is indeed done by the automatic conversion of future. 
And you're right, it isn't necessary here. Lint doesn't seem to complain either, so we can remove this.

**Comment by RobbeSneyders** [comment]
The linter did run before pushing my new changes.
The tests won't run again for some reason.

File: sdks/python/apache_beam/internal/util.py (hunk 1 above)
Comment (tvalentyn) [reply sdks/python/apache_beam/internal/util.py]: Hm... the change is harmless, I totally don't mind it. I am a little concerned that we might not be able to enforce consistency in the codebase, and automatically catch when such changes are required. Say, tomorrow somebody adds a usage of `next` in against an object  passed into the method, and we can't tell if the object will has the iterator implemented in Py3-compliant way, so then there would be a difference in behavior between py2 and py3, and importing `next` would make a difference, but if linter does not catch it, chances are reviewers  won't notice it either.

File: sdks/python/tox.ini [hunk 2]
```diff
@@ -17,7 +17,8 @@
 
 [tox]
 # new environments will be excluded by default unless explicitly added to envlist.
-envlist = py27,py27-{gcp,cython,lint,lint3},py3-lint,docs
+#envlist = py27,py27-{gcp,cython,lint,lint3},py3-lint,docs
```
Comment (aaltay) [inline sdks/python/tox.ini]: Revert this change?

Comment (RobbeSneyders) [reply sdks/python/tox.ini]: My mistake.

File: sdks/python/apache_beam/internal/util.py (hunk 1 above)
Comment (RobbeSneyders) [reply sdks/python/apache_beam/internal/util.py]: I checked again, and it seems like the linter checks for `.next()` calls on objects. So we should be safe.
